<launch>
    <arg name="config" default="vehicle_sim"/>
    <arg name="rosbag" default="false"/>
    <arg name="use_path_follow" default="false"/>

    <!-- 原 path_follower 是为地面机器人设计，不兼容四旋翼 -->
    <group if="$(arg use_path_follow)">
        <include file="$(find iplanner_path_follow)/launch/path_follower.launch" />
    </group>

    <!-- iPlanner 规划节点 -->
    <node pkg="iplanner_node" type="iplanner_node.py" name="iplanner_node" output="screen">
        <rosparam command="load" file="$(find iplanner_node)/config/$(arg config).yaml"/>
    </node>

    <!-- Path to PositionCommand 桥接节点 -->
    <node pkg="iplanner_node" type="path_to_cmd_node.py" name="path_to_cmd_bridge" output="screen">
        <param name="lookahead_dist" value="0.5"/>
        <param name="max_speed" value="2.0"/>
        <param name="max_acc" value="3.0"/>
        <param name="goal_tolerance" value="0.3"/>
        <param name="cmd_freq" value="50.0"/>
        <param name="path_frame" value="base"/>
        <param name="world_frame" value="world"/>
        <param name="target_height" value="1.0"/>
        <param name="takeoff_height" value="1.0"/>
    </node>

    <!-- RViz 可视化 (可选) -->
    <!-- <node type="rviz" name="iplanner_rviz" pkg="rviz" args="-d $(find iplanner_node)/rviz/default.rviz" respawn="false"/> -->
</launch>